packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -0,0 +1,413 @@
+  final String fontFamily;
+
+  /// The ordered list of font families to fall back on when a higher priority
+  /// font family cannot be found.
+  ///
+  /// The value provided in [fontFamily] will act as the preferred/first font
+  /// family that will be searched for, followed in order by the font families
+  /// in [fontFamilyFallback]. If all font families are exhausted and no match
+  /// was found, the default platform font family will be used instead. Unlike
+  /// [TextStyle.fontFamilyFallback], the font does not need to contain the
+  /// desired glyphs to match.
+  ///
+  /// When [fontFamily] is null or not provided, the first value in [fontFamilyFallback]
+  /// acts as the preferred/first font family. When neither is provided, then
+  /// the default platform font will be used. Providing and empty list or null
+  /// for this property is the same as omitting it.
+  ///
+  /// If the font is defined in a package, each font family in the list will be
+  /// prefixed with 'packages/package_name/' (e.g. 'packages/cool_fonts/Roboto').
+  /// The package name should be provided by the `package` argument in the
+  /// constructor.
+  List<String> get fontFamilyFallback => _package != null && _fontFamilyFallback != null ? _fontFamilyFallback.map((String str) => 'packages/$_package/$str').toList() : _fontFamilyFallback;
+  final List<String> _fontFamilyFallback;
+
+  // This is stored in order to prefix the fontFamilies in _fontFamilyFallback
+  // in the [fontFamilyFallback] getter.
+  final String _package;
+
+  /// The minimum size of glyphs (in logical pixels) to use when painting the text.
+  ///
+  /// During painting, the [fontSize] is multiplied by the current
+  /// `textScaleFactor` to let users make it easier to read text by increasing
+  /// its size.
+  ///
+  /// The default fontSize is `14` logical pixels.
+  final double fontSize;
+
+  /// The height of the line as a ratio of fontSize. This property does
+  /// not affect the leading height, which is controlled separately through
+  /// [leading].
+  ///
+  /// The default lineHeight is `1.0`.
+  final double lineHeight;
+
+  /// The typeface thickness to use when calculating the strut (e.g., bold).
+  ///
+  /// The default fontWeight is `w400`.
+  final FontWeight fontWeight;
+
+  /// The typeface variant to use when calculating the strut (e.g., italics).
+  ///
+  /// The default fontStyle is `normal`.
+  final FontStyle fontStyle;
+
+  /// The custom strut leading as a ratio of [fontSize].
+  ///
+  /// If this is null or negative, the font's leading will be used. Positive
+  /// values and zero specifies a custom leading, half of which will be applied
+  /// to the top of the line box and the other half to the bottom.
+  ///
+  /// The default leading is `-1`, which will use the font-specified leading.
+  final double leading;
+
+  /// Whether the strut height should be forced.
+  ///
+  /// When true, all lines will be laid out with the height of the
```

Comment:
I guess it would be more accurate to say that the baselines will be laid out according to the strut, and the rest of the text would blindly place their alphabetic baselines along this forced baseline.